src/commands/set.json

```diff
@@ -0,0 +1,147 @@
+{
+    "SET":{
+        "summary":"Set the string value of a key",
+        "complexity":"O(1)",
+        "since":"1.0.0",
+        "group":"string",
+        "arity":-3,
+        "function":"setCommand",
+        "command_flags":[
+            "write",
+            "use-memory"
+        ],
+        "acl_categories":[
+            "string",
+            "write",
+            "slow"
+        ],
+        "key_specs":[
+            {
+                "begin_search":{
+                    "index":{
+                        "pos":1
+                    }
+                },
+                "find_keys":{
+                    "range":{
+                        "lastkey":0,
+                        "step":1,
+                        "limit":0
+                    }
+                },
+                "flags":[
+                    "write"
+                ]
+            }
+        ],
+        "returns":[
+            {
+                "description": "Command succeeded",
+                "type":"simple-string",
+                "constant_value": "+OK"
+            },
+            {
+                "description": "Old value if `GET` was given",
+                "type":"bulk-string"
+            },
+            {
+                "description": "Either `SET` failed (with `NX` or `XX`), or `GET` was given and key didn't exist",
+                "type": {
+                    "RESP2":"null-bulk-string",
+                    "RESP3":"null"
+                }
+            }
```

Comment:
this looks much better.
i wonder what would be the doc for something like ZPOP
it differs between RESP2 and RESP3 by both the fact the reply is a nested array, but also by the fact the the score is double rather than a string.
i guess this would still be a single entry in the "returns" array, right?
i suppose that for human readers that's enough, but maybe clients would want to further parse it and understand the nested types?

maybe not, i remind myself that the whole purpose of RESP3 is that clients don't need to understand the command in order to process the reply.

Reply:
i think we need to describe just the immediate return-type, and not dive into specifics.. but I'll add it as a question

Reply:
will be handled in https://github.com/redis/redis/issues/9845